### PR TITLE
Invitation letter generator

### PR DESCRIPTION
**Pull request by mthembbu**
#328

### CODE REVIEW

File: api/app/invitationletter/generator.py [hunk 1]
```diff
@@ -1,9 +1,105 @@
...
+
+os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = json.dumps(GCP_CREDENTIALS_DICT)
+
+
+OFFER_EMAIL_BODY = """
+Dear {user_title} {first_name} {last_name},
+
+Congratulations! You've been selected to attend the {event_name}!
+
+Please see the attached document below for your acceptance of offer: {host}/offer
```
Comment (jaderabbit) [inline api/app/invitationletter/generator.py]: I'm not sure if this is github being weird, but I gave specific copy for this
It's not an offer email. it's an invitation letter

File: api/app/invitationletter/generator.py [hunk 2]
```diff
@@ -1,9 +1,105 @@
...
+from app.utils import emailer
+from app.utils import pdfconvertor
+from app.events.models import Event
+from app import db
+from app.utils import errors
+
+os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = json.dumps(GCP_CREDENTIALS_DICT)
+
+
+OFFER_EMAIL_BODY = """
```
Comment (jaderabbit) [inline api/app/invitationletter/generator.py]: it's not an offer email, it's an invitation email

File: api/app/invitationletter/generator.py [hunk 3]
```diff
@@ -1,9 +1,107 @@
...
+        LOGGER.debug("merge-fields.... {} .".format(document.get_merge_fields()))
+
+        document.merge(
+            TITLE=user_title,
+            FIRSTNAME=firstname,
+            SURNAME=lastname,
+            WORK_ADDRESS=work_address,
+            ADDRESSED_TO=addressed_to,
+            RESIDENTIAL_ADDRESS=residential_address,
+            PASSPORT_NAME=passport_name,
```
Comment (KaleabTessera) [inline api/app/invitationletter/generator.py]: I think having ```Firstname```, ```Surname``` and ```Passport_Name``` is redundant.  I think we should just use ```Passport_Name```.

Comment (jaderabbit) [reply api/app/invitationletter/generator.py]: The templates are currently using firstname, lastname. We don't even use passport name, but we should be

Comment (KaleabTessera) [reply api/app/invitationletter/generator.py]: Can't we just replace firstname and lastname, with Passport_name? We just need their full name as  shown on their passport. 

Even on the templates we can just change it to use to Passport_name. I can't think of a use case for having the names split and having seperate fields for firstname,lastname and passport name.

Comment (jaderabbit) [reply api/app/invitationletter/generator.py]: Agreed. Can we merge for now though? Testing the download from gcp is proving to be nightmare, so would like to test on staging if we can.

Comment (KaleabTessera) [reply api/app/invitationletter/generator.py]: Cool.

**Pull request merged**